### PR TITLE
Fix Civil3D hostname to fix psuedo warning for package users

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

https://jira.autodesk.com/browse/DYN-5470
Added `Dynamo Civil 3D` as the hostname for Civil 3D packages that have `Civil 3D` as the host dependency.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoPackages/PackageManagerExtension.cs [hunk 1]
```diff
@@ -44,6 +44,14 @@ public class PackageManagerExtension : IExtension, ILogSource, IExtensionSource
         /// </summary>
         private Dictionary<string, List<PackageInfo>> NodePackageDictionary;
 
+        // TODO : Update all packages to use this hostname and replace the hostname in package manager as well.
+        /// <summary>
+        /// This will be used to match Civil 3D specific packages,
+        /// as there is a mismatch between the host dependency name in packages and the host name used by Civil 3D for Dynamo.
+        /// </summary>
+        private readonly string Civil3DHostName = "Dynamo Civil 3D";
+        private readonly string oldCivil3DHostName = "Civil 3D";
```
Comment (QilongTang) [inline src/DynamoPackages/PackageManagerExtension.cs:53]: hmm it's a bit weird to define this in the Dynamo code base, should we switch to a more tolerant match based on your earlier comment?

Comment (mjkkirschner) [reply src/DynamoPackages/PackageManagerExtension.cs:53]: just curious - could we do this transformation at runtime on the package manager?

Comment (zeusongit) [reply src/DynamoPackages/PackageManagerExtension.cs:53]: @mjkkirschner Do you mean adding the hostname in the hosts returned from Package Manager?
here: https://git.autodesk.com/Dynamo/PackageManager/blob/c1e6047b5a92419049098c4a478bfb463aed6aae/src/app.js#L239

Comment (zeusongit) [reply src/DynamoPackages/PackageManagerExtension.cs]: > 

updated with more tolerant approach

Comment (mjkkirschner) [reply src/DynamoPackages/PackageManagerExtension.cs]: no, I mean that when package manager encounters a package with this old hostname to just return the new one instead in the response when the user downloads this package or gets its header... but perhaps this has issues with the pkg.json?

**Comment by reddyashish** [comment]
looks like a sporadic failure. Running the job again: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/8921/

**Comment by reddyashish** [comment]
Cherrypick this?

**Pull request merged**